docs/api/ref/GetAgent.md

```diff
@@ -0,0 +1,224 @@
+
+<Tabs groupId="lang">
+<TabItem value="ts" label="TypeScript">
+
+```ts
+import { getAgent, DesktopAgent } from "@finos/fdc3";
+
+try {
+    const desktopAgent: DesktopAgent = await getAgent();
+    //do FDC3 things here
+} catch (e: AgentError) {
+    // Failed to connect
+}
+```
+
+</TabItem>
+<TabItem value="js" label="JavaScript">
+
+```js
+import { getAgent } from "@finos/fdc3";
+
+try {
+    const desktopAgent = await getAgent();
+    //do FDC3 things here
+} catch (e) {
+    // Failed to connect
+}
+```
+
+</TabItem>
+</Tabs>
+
+The  `getAgent()` function allows web applications to retrieve an FDC3 Desktop Agent API interface to work with, whether they are running in an environment that supports a Desktop Agent Preload (a container-injected API implementation) or a Desktop Agent Proxy (a Browser-based Desktop Agent running in another window or frame). The behavior of `getAgent()` is defined by the [FDC3 Web Connection Protocol (WCP)](../specs/webConnectionProtocol) and communication with a Desktop Agent Proxy in a web-browser is defined by the [Desktop Agent Communication Protocol (DACP)](../specs/desktopAgentCommunicationProtocol). Hence, it allows applications to be written that will work in either scenario without modification or the inclusion of vendor-specific libraries.
+
+If no Desktop Agent is found, a failover function may be supplied by app allowing it to start or otherwise connect to a Desktop Agent (e.g. by loading a proprietary adaptor that returns a `DesktopAgent` implementation or by creating a window or iframe of its own that will provide a Desktop Agent Proxy).
+
+The definition of the `getAgent()` function is as follows:
+
+```ts
+type GetAgentType = ( 
+  params?: GetAgentParams,  
+) => Promise<DesktopAgent>; 
+```
+
+A small number of arguments are accepted that can affect the behavior of `getAgent` and to provide a fallback in case a Desktop Agent is not found, allowing the application to start its own agent or use another mechanism (such a proprietary adaptor) to connect to one.
+
+```ts
+/** 
+ * @typedef {Object} GetAgentParams Type representing parameters passed to the 
+ * getAgent function. 
+ * 
+ * @property {string} identityUrl The app's current URL is normally sent to 
+ * a web-based desktop agent to help establish its identity. This property 
+ * may be used to override the URL sent (to handle situations where an app's 
+ * URL is not sufficiently stable to use for identity purposes). The URL set 
+ * MUST match the origin of the application (scheme, hostname, and port) or  
+ * it will be ignored. If not specified, the app's current URL will be used.  
+ * 
+ * @property {number} timeout Number of milliseconds to allow for an fdc3  
```

Comment:
The old fdc3Ready function had an argument named with an "Ms" suffix ("waitForMs").  So it's not clear to me why this param is named simply "timeout" instead of being named "timeoutMs".

Apologies if I missed a justification during previous discussions - perhaps some people hated the suffix, and preferred to deviate from what was done previously with fdc3Ready?

I appreciate one could argue the suffix only helps developers who refuse to RTFM :)

Reply:
Changed to timeoutMs in be131b655bbc4812c9cfec080fb8ee86ac81cb85

WIll require a code change @robmoffat - should be highlighted for you if importing the `GetAgent` type